src/super_gradients/examples/qat_from_recipe_example/qat_from_recipe.py

```diff
@@ -0,0 +1,239 @@
+        cfg.dataset_params.train_dataloader_params.batch_size //= cfg.quantization_params.modify_recipe_for_qat.batch_size_divisor
+        cfg.dataset_params.val_dataloader_params.batch_size //= cfg.quantization_params.modify_recipe_for_qat.batch_size_divisor
+
+        logger.warning(f"New dataset_params.train_dataloader_params.batch_size: {cfg.dataset_params.train_dataloader_params.batch_size}")
+        logger.warning(f"New dataset_params.val_dataloader_params.batch_size: {cfg.dataset_params.val_dataloader_params.batch_size}")
+
+    cfg.training_hyperparams.max_epochs //= cfg.quantization_params.modify_recipe_for_qat.max_epochs_divisor
+    logger.warning(f"New number of epochs: {cfg.training_hyperparams.max_epochs}")
+
+    lr_decay_factor = cfg.quantization_params.modify_recipe_for_qat.lr_decay_factor
+
+    cfg.training_hyperparams.initial_lr *= lr_decay_factor
+    if cfg.training_hyperparams.warmup_initial_lr is not None:
+        cfg.training_hyperparams.warmup_initial_lr *= lr_decay_factor
+    else:
+        cfg.training_hyperparams.warmup_initial_lr = cfg.training_hyperparams.initial_lr * 0.01
+
+    cfg.training_hyperparams.optimizer_params.weight_decay *= lr_decay_factor
+
+    logger.warning(f"New learning rate: {cfg.training_hyperparams.initial_lr}")
+    logger.warning(f"New weight decay: {cfg.training_hyperparams.optimizer_params.weight_decay}")
+
+    # as recommended by pytorch-quantization docs
+    cfg.training_hyperparams.lr_mode = "cosine"
+    cfg.training_hyperparams.lr_warmup_epochs = (
+        cfg.training_hyperparams.max_epochs // cfg.quantization_params.modify_recipe_for_qat.warmup_epochs_divisor
+    ) or 1
+    cfg.training_hyperparams.cosine_final_lr_ratio = cfg.quantization_params.modify_recipe_for_qat.cosine_final_lr_ratio
+
+    # do mess with Q/DQ
+    if cfg.training_hyperparams.ema:
+        logger.warning("EMA will be disabled for QAT run.")
+        cfg.training_hyperparams.ema = False
+
+    if cfg.training_hyperparams.sync_bn:
+        logger.warning("SyncBatchNorm will be disabled for QAT run.")
+        cfg.training_hyperparams.sync_bn = False
+
+    if cfg.quantization_params.modify_recipe_for_qat.disable_phase_callbacks and len(cfg.training_hyperparams.phase_callbacks) > 0:
+        logger.warning(f"Recipe contains {len(cfg.training_hyperparams.phase_callbacks)} phase callbacks. All of them will be disabled.")
+        cfg.training_hyperparams.phase_callbacks = []
+
+    if cfg.multi_gpu != "OFF" or cfg.num_gpus != 1:
+        logger.warning(f"Recipe requests multi_gpu={cfg.multi_gpu} and num_gpus={cfg.num_gpus}. Changing to multi_gpu=OFF and num_gpus=1")
+        cfg.multi_gpu = "OFF"
+        cfg.num_gpus = 1
+
+    # no augmentations
+    if "transforms" in cfg.dataset_params.val_dataset_params:
+        cfg.dataset_params.train_dataset_params.transforms = cfg.dataset_params.val_dataset_params.transforms
+
+
+def qat_from_config(cfg):
```

Comment:
See my point regarding inheriting from SgTrainer and modifying train_from_config.